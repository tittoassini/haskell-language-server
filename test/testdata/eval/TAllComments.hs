-- Tests can be in either plain or Haddock comments in both single line or multi line format.
-- Comments and tests that do not start on the first column or the first line of a multi-line comment are ignored.
module AllComments where

-- Single line plain comment
-- >>> "a"++"b"

{- Multi line plain comment
>>> "b"++"c"
-}

-- | Single line Haddock comment
-- >>> "a"++"b"

{- Multi line Haddock comment
>>> "b"++"c"
-}

twice :: [a] -> [a]
twice a = a ++ a
-- ^ Also in backward Haddock comments
-- >>> twice "ABC"

   -- Ignored as it doesn't start on the first column.
   -- >>> IGNORED

{- | >>> 2+5

     >>> IGNORED
-}
