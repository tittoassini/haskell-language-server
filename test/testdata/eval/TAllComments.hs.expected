-- Tests can be in either plain or Haddock comments in both single line or multi line format.
-- Comments and tests that do not start on the first column or the first line of a multi-line comment are ignored.
module TAllComments where

-- Single line plain comment
-- >>> "a"++"b"
-- "ab"

{- Multi line plain comment
>>> "b"++"c"
"bc"
-}

{- | Single line Haddock comment
 >>> "a"++"b"
-}

{- Multi line Haddock comment
>>> "b"++"c"
"bc"
-}

twice :: [a] -> [a]
twice a = a ++ a
{- ^ Also in backward Haddock comments
 >>> twice "ABC"
-}

-- Ignored as it doesn't start on the first column.
-- >>> IGNORED
-- Data constructor not in scope: IGNORED

{- | >>> 2+5
7

     >>> IGNORED
-}
